src/components/reusable/details/details.ts

```diff
@@ -0,0 +1,75 @@
+import { LitElement, html } from 'lit';
+import { customElement, property, state, query } from 'lit/decorators.js';
+import Styles from './details.scss';
+
+/**
+ * Example component template.
+ * @fires on-click - Captures the click event and emits a custom event.
+ * @slot unnamed - Slot for child components.
+ */
+@customElement('kyn-details')
+export class Details extends LitElement {
+  static override styles = Styles;
+
+  /** exposed reactive string property */
+  @property({ type: String })
+  summary = '';
+
+  /** exposed reactive array property */
+  @property({ type: Array })
+  details: Array<any> = [];
+
+  /** internal reactive property
+   * @internal
+   */
+  @state()
+  _internalProp = 'Internal Prop'; // use an underscore to signify internal variables/methods
+
+  /** .component element reference. does not get updated until after Lit update lifecycle completes
+   * @internal
+   */
+  @query('.component')
+  _component!: HTMLElement;
```

Comment:
Make sure to remove anything unused when copying from the template or another component.